src/js/common/schemaform/reducers/index.js

```diff
@@ -17,16 +17,30 @@ import { SET_DATA,
 } from '../actions';
 
 function recalculateSchemaAndData(initialState) {
-  return Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
+  const retVal = Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = Object.keys(state).reduce((carry, pageName) => {
```

Comment:
As far as speed goes, they're close enough (and fast enough) it doesn't really make a difference, but not using the object spread is quicker. Taking out the `.filter()` speeds it up, but only by about 0.02ms.

It just comes down to readability. Which is clearer? Does it really matter with the explanatory comment?